common/src/main/java/mod/chiselsandbits/block/entities/ChiseledBlockEntity.java

```diff
@@ -108,6 +109,28 @@ public ChiseledBlockEntity(BlockPos position, BlockState state)
                           .buildMultiThreaded();
     }
 
+    public void updateModelData()
+    {
+        ChiseledBlockModelDataManager.getInstance().updateModelData(this);
+    }
+
+    private void updateModelDataIfInLoadedChunk()
+    {
+        if (level != null && level.isClientSide() && level.isLoaded(getBlockPos()))
+            updateModelData();
+    }
+
+    public static void updateAllModelDataInChunk(LevelChunk chunk)
+    {
+        chunk.getBlockEntities()
+                .values()
+                .forEach(blockEntity ->
+                {
+                    if (blockEntity instanceof ChiseledBlockEntity chiseledBlockEntity)
+                        chiseledBlockEntity.updateModelData();
+                });
+    }
+
```

Comment:
Please move this method to a seperate handler in the logic packet, since this is an event callback.